### PR TITLE
Make checkout e2e tests viewport aware

### LINKED ISSUE #11341: test: Make checkout e2e tests viewport aware
Subtask of #11163

### PR DESCRIPTION
**Pull request by hackergil**
Fixes #11341

### CODE REVIEW

File: projects/storefrontapp-e2e-cypress/cypress/helpers/checkout-as-persistent-user.ts [hunk 1]
```diff
@@ -321,51 +321,18 @@ export function deletePaymentCard() {
 }
 
 export function checkoutAsPersistentUserTest() {
```
Comment (marlass) [inline projects/storefrontapp-e2e-cypress/cypress/helpers/checkout-as-persistent-user.ts]: This is not used. Can we remove this function?

File: projects/storefrontapp-e2e-cypress/cypress/helpers/checkout-flow.ts [hunk 2]
```diff
@@ -22,15 +22,48 @@ export const ELECTRONICS_CURRENCY = 'USD';
 
 export const ELECTRONICS_DEFAULT_DELIVERY_MODE = 'deliveryMode-standard-net';
 
+/**
+ * Clicks the main menu (on mobile only)
+ */
+export function clickHamburger() {
```
Comment (marlass) [inline projects/storefrontapp-e2e-cypress/cypress/helpers/checkout-flow.ts:28]: We have the same util in homepage helpers. Maybe just use the cy.OnMobile there and drop this one?

File: projects/storefrontapp-e2e-cypress/cypress/helpers/checkout-as-persistent-user.ts (hunk 1 above)
Comment (hackergil) [reply projects/storefrontapp-e2e-cypress/cypress/helpers/checkout-as-persistent-user.ts]: It is used by a few e2es. We can remove it when we get to remove those dependencies I would say.

File: projects/storefrontapp-e2e-cypress/cypress/helpers/checkout-flow.ts (hunk 2 above)
Comment (hackergil) [reply projects/storefrontapp-e2e-cypress/cypress/helpers/checkout-flow.ts:28]: It's a one liner and I would like to remove dependencies to other helpers as much as possible until we are in a better position to reference them. What do you think?

File: projects/storefrontapp-e2e-cypress/cypress/helpers/checkout-as-persistent-user.ts (hunk 1 above)
Comment (marlass) [reply projects/storefrontapp-e2e-cypress/cypress/helpers/checkout-as-persistent-user.ts]: Which e2e? I don't see any

Comment (hackergil) [reply projects/storefrontapp-e2e-cypress/cypress/helpers/checkout-as-persistent-user.ts]: My bad. Removed.

**Review by marlass — APPROVED** [review]

**Pull request merged**